gensim/models/atmodel.py

```diff
@@ -0,0 +1,945 @@
+            authors_d = [self.author2id[a] for a in self.doc2author[doc_no]]
+            ids = np.array([id for id, _ in doc])  # Word IDs in doc.
+            cts = np.array([cnt for _, cnt in doc])  # Word counts.
+
+            if d % self.chunksize == 0:
+                logger.debug("bound: at document #%i in chunk", d)
+
+            # Computing the bound requires summing over expElogtheta[a, k] * expElogbeta[k, v], which
+            # is the same computation as in normalizing phi.
+            phinorm = self.compute_phinorm(ids, authors_d, expElogtheta[authors_d, :], expElogbeta[:, ids])
+            word_score += np.log(1.0 / len(authors_d)) + cts.dot(np.log(phinorm))
+
+        # Compensate likelihood for when `chunk` above is only a sample of the whole corpus. This ensures
+        # that the likelihood is always rougly on the same scale.
+        word_score *= subsample_ratio
+
+        # E[log p(theta | alpha) - log q(theta | gamma)]
+        for a in author2doc.keys():
+            a = self.author2id[a]
+            theta_score += np.sum((self.alpha - gamma[a, :]) * Elogtheta[a, :])
+            theta_score += np.sum(gammaln(gamma[a, :]) - gammaln(self.alpha))
+            theta_score += gammaln(np.sum(self.alpha)) - gammaln(np.sum(gamma[a, :]))
+
+        # theta_score is rescaled in a similar fashion.
+        # TODO: treat this in a more general way, similar to how it is done with word_score.
+        theta_score *= self.num_authors / len(author2doc)
+
+        # E[log p(beta | eta) - log q (beta | lambda)]
+        beta_score = 0.0
+        beta_score += np.sum((self.eta - _lambda) * Elogbeta)
+        beta_score += np.sum(gammaln(_lambda) - gammaln(self.eta))
+        sum_eta = np.sum(self.eta)
+        beta_score += np.sum(gammaln(sum_eta) - gammaln(np.sum(_lambda, 1)))
+
+        total_score = word_score + theta_score + beta_score
+
+        return total_score
+
+    def get_document_topics(self, word_id, minimum_probability=None):
+        '''
+        This method overwrites `LdaModel.get_document_topics` and simply raises an
+        exception. `get_document_topics` is not valid for the author-topic model,
+        use `get_author_topics` instead.
+
+        '''
+
+        raise NotImplementedError('Method "get_document_topics" is not valid for the author-topic model. Use the "get_author_topics" method.')
+
+    def get_author_topics(self, author_name, minimum_probability=None):
+        """
+        Return topic distribution the given author, as a list of
+        (topic_id, topic_probability) 2-tuples.
+        Ignore topics with very low probability (below `minimum_probability`).
+
+        Obtaining topic probabilities of each word, as in LDA (via `per_word_topics`),
+        is not supported.
+
+        """
+
+        author_id = self.author2id[author_name]
+
+        if minimum_probability is None:
+            minimum_probability = self.minimum_probability
+
+        # NOTE: this is used in LdaModel:
```

Comment:
Well. I was really sceptical about doing that. But now I just decided to do it anyway. So the note is removed and the line is uncommented (so it is doing what LdaModel is doing now).